crypto/digestible/derive/test/tests/behavior.rs

```diff
@@ -67,38 +42,81 @@ struct GenericFoo<X: Digestible> {
 #[test]
 fn foo1() {
     let arg = Foo { a: 0, b: 1, c: 2 };
-    let mut hasher = Tester::new();
-    arg.digest(&mut hasher);
-
-    let expected: Vec<Vec<u8>> = vec![
-        b"Foo".to_vec(),
-        b"".to_vec(),
-        b"a".to_vec(),
-        vec![0u8, 0u8],
-        b"b".to_vec(),
-        vec![1u8, 0u8],
-        b"c".to_vec(),
-        vec![2u8, 0u8],
-    ];
-
-    assert_eq!(hasher.args, expected);
+    let expected_ast = ASTNode::from(ASTAggregate {
+        context: b"foo1",
+        name: b"Foo".to_vec(),
+        elems: vec![
+            ASTNode::from(ASTPrimitive {
+                context: b"a",
+                type_name: b"uint",
+                data: vec![0u8, 0u8],
+            }),
+            ASTNode::from(ASTPrimitive {
+                context: b"b",
+                type_name: b"uint",
+                data: vec![1u8, 0u8],
+            }),
+            ASTNode::from(ASTPrimitive {
+                context: b"c",
+                type_name: b"uint",
+                data: vec![2u8, 0u8],
+            }),
+        ],
+        is_completed: true,
+    });
+    digestible_test_case_ast("foo1", &arg, expected_ast);
+    assert_eq!(
+        arg.digest32::<MerlinTranscript>(b"foo1"),
+        [
+            44, 174, 200, 201, 207, 72, 151, 193, 63, 97, 49, 112, 135, 50, 252, 61, 211, 208, 41,
+            4, 241, 249, 90, 114, 181, 197, 177, 151, 39, 165, 76, 197
+        ]
+    );
 }
 
 #[test]
 fn blob1() {
```

Comment:
A comment here like "A tuple struct should be treated as an Aggregate" would help make it clear how this test and the following test about TransBlob should differ.